Build failure with Hasql 1.3

```
src/Hasql/Migration/Util.hs:19:1: error:
    Could not find module ‘Hasql.Query’
    Use -v to see a list of the files searched for.
   |
19 | import           Hasql.Query
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Hasql/Migration/Util.hs:22:37: error:
    Module ‘Hasql.Transaction’ does not export ‘query’
   |
22 | import           Hasql.Transaction (query, Transaction)
   |  
```

Hasql 1.3 has dropped the `Query` module in favor of a `Statement` module, which unfortunately causes `hasql-migration` to fail to build with the error above.

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,7 +7,6 @@ before_script:
  - psql -c 'create database test;' -U postgres
 
 env:
- - CABALVER=1.22 GHCVER=7.10.3
  - CABALVER=1.24 GHCVER=8.0.1
  - CABALVER=2.0 GHCVER=8.2.2
  - CABALVER=2.2 GHCVER=8.4.3

--- a/.travis.yml
+++ b/.travis.yml
@@ -10,6 +10,7 @@ env:
  - CABALVER=1.22 GHCVER=7.10.3
  - CABALVER=1.24 GHCVER=8.0.1
  - CABALVER=2.0 GHCVER=8.2.2
+ - CABALVER=2.2 GHCVER=8.4.3
 
 before_install:
  - travis_retry sudo add-apt-repository -y ppa:hvr/ghc

--- a/hasql-migration.cabal
+++ b/hasql-migration.cabal
@@ -1,5 +1,5 @@
 name:                       hasql-migration
-version:                    0.1.4
+version:                    0.2.0
 synopsis:                   PostgreSQL Schema Migrations
 homepage:                   https://github.com/tvh/hasql-migration
 Bug-reports:                https://github.com/tvh/hasql-migration/issues
@@ -38,8 +38,8 @@ library
                             cryptonite                  >= 0.11,
                             data-default-class          >= 0.0.1,
                             directory                   >= 1.2,
-                            hasql                       >= 0.19,
-                            hasql-transaction           >= 0.5,
+                            hasql                       >= 1.3,
+                            hasql-transaction           >= 0.7,
                             memory,
                             text                        >= 1.2,
                             time                        >= 1.4

--- a/hasql-migration.cabal
+++ b/hasql-migration.cabal
@@ -12,7 +12,7 @@ category:                   Database
 build-type:                 Simple
 cabal-version:              >= 1.10
 description:                A PostgreSQL-simple schema migration utility
-
+tested-with:                GHC==7.10.3, GHC==8.0.1, GHC==8.2.2, GHC==8.4.3
 extra-source-files:         License
                             Readme.markdown
 

--- a/hasql-migration.cabal
+++ b/hasql-migration.cabal
@@ -12,7 +12,7 @@ category:                   Database
 build-type:                 Simple
 cabal-version:              >= 1.10
 description:                A PostgreSQL-simple schema migration utility
-tested-with:                GHC==7.10.3, GHC==8.0.1, GHC==8.2.2, GHC==8.4.3
+tested-with:                GHC==8.0.1, GHC==8.2.2, GHC==8.4.3
 extra-source-files:         License
                             Readme.markdown
 

--- a/src/Hasql/Migration.hs
+++ b/src/Hasql/Migration.hs
@@ -43,7 +43,7 @@ import Data.List (isPrefixOf, sort)
 import Data.Time (LocalTime)
 import Data.Traversable (forM)
 import Hasql.Migration.Util (existsTable)
-import Hasql.Query
+import Hasql.Statement
 import Hasql.Transaction
 import System.Directory (getDirectoryContents)
 import qualified Data.ByteString as BS (ByteString, readFile)
@@ -95,7 +95,7 @@ executeMigration name contents = do
             return Nothing
         ScriptNotExecuted -> do
             sql contents
-            query (name, checksum) (statement q (contramap (first T.pack) def) Decoders.unit False)
+            statement (name, checksum) (Statement q (contramap (first T.pack) def) Decoders.unit False)
             return Nothing
         ScriptModified _ -> do
             return (Just $ ScriptChanged name)
@@ -147,7 +147,7 @@ executeValidation cmd = case cmd of
 -- will be executed and its meta-information will be recorded.
 checkScript :: ScriptName -> Checksum -> Transaction CheckScriptResult
 checkScript name checksum =
-    query name (statement q (contramap T.pack (Encoders.value def)) (Decoders.maybeRow (Decoders.value def)) False) >>= \case
+    statement name (Statement q (contramap T.pack (Encoders.param def)) (Decoders.rowMaybe (Decoders.column def)) False) >>= \case
         Nothing ->
             return ScriptNotExecuted
         Just actualChecksum | checksum == actualChecksum ->
@@ -201,7 +201,7 @@ data MigrationError = ScriptChanged String | NotInitialised | ScriptMissing Stri
 -- | Produces a list of all executed 'SchemaMigration's.
 getMigrations :: Transaction [SchemaMigration]
 getMigrations =
-    query () $ statement q def (Decoders.rowsList decodeSchemaMigration) False
+    statement () $ Statement q def (Decoders.rowList decodeSchemaMigration) False
     where
         q = mconcat
             [ "select filename, checksum, executed_at "
@@ -225,6 +225,6 @@ instance Ord SchemaMigration where
 decodeSchemaMigration :: Decoders.Row SchemaMigration
 decodeSchemaMigration =
     SchemaMigration
-    <$> Decoders.value def
-    <*> Decoders.value def
-    <*> Decoders.value def
+    <$> Decoders.column def
+    <*> Decoders.column def
+    <*> Decoders.column def

--- a/src/Hasql/Migration/Util.hs
+++ b/src/Hasql/Migration/Util.hs
@@ -16,17 +16,17 @@ module Hasql.Migration.Util
     ( existsTable
     ) where
 
-import           Hasql.Query
+import           Hasql.Statement
 import qualified Hasql.Encoders as Encoders
 import qualified Hasql.Decoders as Decoders
-import           Hasql.Transaction (query, Transaction)
+import           Hasql.Transaction (statement, Transaction)
 import Data.Text (Text)
 import Data.Default.Class
 
 -- | Checks if the table with the given name exists in the database.
 existsTable :: Text -> Transaction Bool
 existsTable table =
-    fmap (not . null) $ query table q
+    fmap (not . null) $ statement table q
     where
-        q = statement sql (Encoders.value def) (Decoders.rowsList (Decoders.value Decoders.int8)) False
+        q = Statement sql (Encoders.param def) (Decoders.rowList (Decoders.column Decoders.int8)) False
         sql = "select relname from pg_class where relname = $1"

--- a/test/Hasql/MigrationTest.hs
+++ b/test/Hasql/MigrationTest.hs
@@ -14,7 +14,7 @@
 
 module Hasql.MigrationTest where
 
-import           Hasql.Session                        (run, Error)
+import           Hasql.Session                        (run, QueryError)
 import           Hasql.Connection
 import qualified Hasql.Transaction                    as Tx
 import qualified Hasql.Transaction.Sessions           as Tx
@@ -23,7 +23,7 @@ import           Hasql.Migration.Util                 (existsTable)
 import           Test.Hspec                           (Spec, describe, it,
                                                        shouldBe, runIO)
 
-runTx :: Connection -> Tx.Transaction a -> IO (Either Error a)
+runTx :: Connection -> Tx.Transaction a -> IO (Either QueryError a)
 runTx con act = do
     run (Tx.transaction Tx.ReadCommitted Tx.Write act) con
 

--- a/test/Main.hs
+++ b/test/Main.hs
@@ -18,7 +18,7 @@ module Main
 
 import Hasql.Connection
 import Hasql.MigrationTest
-import           Test.Hspec                               (hspec)
+import Test.Hspec (hspec)
 
 main :: IO ()
 main = do

